provisioning/device/devdoc/client_state_machine_requirements.md

```diff
@@ -46,12 +38,12 @@ Register round-trips one step of the registration process, not returning until a
 **SRS_NODE_PROVISIONING_TRANSPORT_STATE_MACHINE_18_024: [** If `register` is called while a different request is in progress, it shall fail with an `InvalidOperationError`. **]**
 
 
-### disconnect(callback: (err: Error) => void): void
-Disconnect is used to disconnect the transport
+### endSession(callback: (err: Error) => void): void
```

Comment:
>endSession [](start = 4, length = 10)

during other SDKs reviews we talked about this and I believe we ended up deciding that we should automatically "end the session" when we get to the assigned state.

as to how to provide a way to "end the session" before it's complete, TBD. 

in any case, we should not rely on the customer to call this to clean things up when registration is done.

Reply:
OK.  We may still want to be explicit at this level though.

---
In reply to: [148916208](https://github.com/Azure/azure-iot-sdk-node/pull/135#discussion_r148916208) [](ancestors = 148916208)